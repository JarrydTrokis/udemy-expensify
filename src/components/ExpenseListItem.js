import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import { toGBP } from '../helpers/toGBP'

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
      {toGBP(amount)}
      -
      {moment(createdAt).format('Do MMMM, YYYY')}
    </p>
  </div>
)

export default ExpenseListItem
