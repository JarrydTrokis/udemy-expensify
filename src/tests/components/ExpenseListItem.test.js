import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'
import { singleExpense } from '../fixtures/expenses'

test('Should render a single Expense List Item', () => {
  const wrapper = shallow(<ExpenseListItem {...singleExpense} />)
  expect(wrapper).toMatchSnapshot()
})
