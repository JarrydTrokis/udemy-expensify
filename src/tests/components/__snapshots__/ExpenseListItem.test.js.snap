// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a single Expense List Item 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/4"
  >
    <h3>
      food shopping
    </h3>
  </Link>
  <p>
    Â£30.00
    -
    2nd January, 1970
  </p>
</div>
`;
